### DISCUSSION TITLE
Support for semantic versioning

### DISCUSSION THREAD

**Discussion by davidsbond** [amacneil/dbmate#371] [Feature Ideas]
Like other database migration tools, `dbmate` only supports numerical version numbers. I would like to be able to version my migrations semantically, using tags like `1.0.0` for example. 

Golang libraries exist that allow you to compare semantic versions, like https://github.com/hashicorp/go-version.

Is there any chance of supporting this style of versioning in future? Is there a database limitation that means these can't be used?

**Comment by amacneil** [comment]
Actually I like this suggestion, I think it would be easy to do and I've been meaning to implement it, so will keep this issue open!
